xdsl/dialects/llvm.py

```diff
@@ -1735,56 +1735,142 @@ def __init__(self, val: Operation | SSAValue, res_type: Attribute):
         )
 
 
+class AbstractFloatArithOp(IRDLOperation, ABC):
+    T: ClassVar = VarConstraint("T", AnyFloatConstr)
+
+    lhs = operand_def(T)
+    rhs = operand_def(T)
+    res = result_def(T)
+
+    fastmathFlags = prop_def(FastMathAttr, default_value=FastMathAttr(None))
+
+    traits = traits_def(Pure())
```

Comment:
Could you please use the`SameOperandsAndResultType` trait where applicable?
It shouldn't be here, as not all ops that inherit from this base should comply to it.